### PR TITLE
Migrate ListView from patternfly-react

### PR DESCRIPTION
**Pull request by rhamilto**
Resolves https://jira.coreos.com/browse/CONSOLE-1721

Note this is a selective migration as patternfly-react's ListView has a significant amount of functionality we're not using, so I've removed anything not in use.  In short, I've copied over the relevant ListView files and removed unneeded functionality.

### CODE REVIEW

**Comment by spadgett** [comment]
/hold
for 4.4

**Comment by rhamilto** [comment]
It occurs to me I should use PatternFly source files and not dist.  Will update.

**Comment by rhamilto** [comment]
> It occurs to me I should use PatternFly source files and not dist. Will update.

Done.

File: frontend/public/components/overview/_project-overview.scss [hunk 1]
```diff
@@ -49,16 +49,6 @@
 
   .project-overview__item {
     min-height: 71px;
-
-    &.list-group-item {
```
Comment (spadgett) [inline frontend/public/components/overview/_project-overview.scss]: nice to be able to remove the overrides 👍

**Comment by rhamilto** [comment]
Refactored to use TypeScript to resolve test errors.

**Comment by rhamilto** [comment]
/hold cancel

@spadgett, PTAL.

**Pull request merged**